packages/buidler-core/src/internal/core/params/argumentTypes.ts

```diff
@@ -159,5 +223,57 @@ export const json: ArgumentType<any> = {
         error
       );
     }
+  },
+  /**
+   *   'json' value validation succeeds if it is of "object" map-like {} type,
```

Comment:
After a lot of thought, I ended up just adding the array case, which is reasonable. But for example I'd say that single values such as `null` or `1234` or `new RegExp()` should not be valid, even if they are JSON.parse-able . 

Part of this decision came from here: https://stackoverflow.com/a/52799327/6279385, 
> JSON specification indicates JSON is built on on two structures: A collection of name/value pair (object) or an ordered list of values (array).

Done in commit bd43dfb.

Also, just ruling out `undefined` is a bit pointless, since the previous logic already ignores undefined values - we have no way of distinguishing if it's undefined due to a lack of definition (and in that case we for example check for defaults), or if it's actually specified as `undefined` value?